apiManager.ts

```diff
@@ -0,0 +1,37 @@
+class ApiManager {
+	
+	private mainUrl: string;
+
+	constructor(mainUrl: string) {
+		this.mainUrl = mainUrl;
+	}
+
+	private fetchJson(url: string): Promise<any> {
+		return fetch(url)
+			.then(res => {
+				if (res.ok) {
+					return res.json()
+				} else {
+					throw new Error(`HTTP error! Status: ${res.status}`)
+				}
+			})
+			.catch(error => {
+				throw new Error(`Fetch failed: ${error}`);
+			});
+	}
+
+	/** Retrieves records from the api */
+	getRecords(fromID: number, toID: number): Promise<string[][]> {
+		return this.fetchJson(`${this.mainUrl}records?from=${fromID}&to=${toID}`);
+	}
+
+	/** Retrieves columns from the api */
+	getColumns(): Promise<string[]> {
+		return this.fetchJson(`${this.mainUrl}columns`);
+	}
+
+	/** Retrieves the number of records there are */
+	getRecordCount(): Promise<number> {
```

Comment:
I seriously want to ask you to cache the record count result.

It is the type of call you can make once and trust that it won't change again.
It is a different case if you are working with live data but in your case, you are not.
I would've made that assumption. You can make the fetch once and save it when this class gets instantiated. Then return the result with `getRecordCount` when you need it. Less chatting to your backend.

However, there is nothing wrong with what you did, as this is a good approach in some cases.